app/src/main/java/com/example/playlistmaker/ui/media/fragments/FavoriteTracksFragment.kt

```diff
@@ -25,13 +30,35 @@ class FavoriteTracksFragment : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        viewModel.state.observe(viewLifecycleOwner) {
-            binding?.llEmptyMediaList?.isVisible = it.isEmpty()
-        }
+        initFavoriteTracksRecyclerView()
+        initObservers()
     }
 
     override fun onDestroyView() {
         super.onDestroyView()
         binding = null
     }
+
+    private fun initFavoriteTracksRecyclerView() {
+        trackAdapter = TrackAdapter(viewModel::clickTrack)
+        binding?.rvFavoriteTracks?.adapter = trackAdapter
+    }
+
+    private fun initObservers() {
+        viewModel.state.observe(viewLifecycleOwner) {
+            trackAdapter?.updateData(it)
+            binding?.apply {
+                noFavoriteTracks.isVisible = it.isEmpty()
+                rvFavoriteTracks.isVisible = it.isNotEmpty()
+            }
+        }
+
+        viewModel.event.observe(viewLifecycleOwner) {
+            when (it) {
+                is FavoriteTracksScreenEvent.OpenPlayerScreen -> {
+                    startActivity(Intent(requireContext(), PlayerActivity()::class.java))
```

Comment:
```suggestion
                    startActivity(Intent(requireContext(), PlayerActivity::class.java))
```
Здесь не стоит создавать активити, чтобы получить ссылку на `Class`

Reply:
исправлено